File: neo4j/perl/read/neo4j_read.pl
#! /usr/bin/perl
#
#	read/neo4j_read.pl
#
#					Aug/30/2012
# -------------------------------------------------------
use	strict;
use	warnings;
use	utf8;
use	Encode;
use	JSON;
#
use lib '/var/www/data_base/common/perl_common';
use json_out;
use uri_get;
#
# -------------------------------------------------------
sub row_parser
{
	my $key = $_[0];

	my $uri = 'http://localhost:7474/db/data/node/' . $key . '/properties';
#
	my $str_json = uri_get::uri_get_proc ($uri);
#
#	print ($str_json);
#
	my $data= decode_json ($str_json);
#
#	if (exists $data{'id'})
		{
	my $out_str = $data->{id} . "\t" . $data->{'name'} . "\t";
	$out_str .= $data->{population} . "\t" . $data->{'date_mod'} . "\n";

	print (encode ('utf-8',$out_str));
		}
}

# -------------------------------------------------------
print	(encode ('utf-8',"*** 開始 ***\n"));
#
for (my $key = 1; $key < 10; $key++)
	{
	row_parser ($key);
	}

#
print	(encode ('utf-8',"*** 終了 ***\n"));
#
# -------------------------------------------------------

File: neo4j/ruby/read/neo4j_read.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
#
#	neo4j_read.rb
#
#					Aug/30/2012
#
require 'restclient'
#
require 'json'
#
# ---------------------------------------------------------------------
def	row_parser (key)
	url_in = 'http://localhost:7474/db/data/node/' + key.to_s + '/properties'
	str_json = RestClient.get url_in
#	puts str_json
#
	hhh=JSON.parse(str_json)
#
	if (hhh.key?("id"))
		print hhh["id"],"\t",hhh["name"],"\t",hhh["population"], \
		"\t",hhh["date_mod"]
		puts
	end
end

#
# ---------------------------------------------------------------------
puts	"*** 開始 ***"
#
for key in 1..9
#	puts key
	row_parser(key)
end
#
puts	"*** 終了 ***"
#
# ---------------------------------------------------------------------
